docs/README-AWS.md

```diff
@@ -398,70 +479,99 @@ inputs:
+So the Lambda function won't return a failure for errors during ingestion; instead, the payload of the event that failed to be ingested will be sent to a replay SQS queue. The replay SQS queue is automatically set up by the Lambda deployment automatically. The replay SQS queue is not set as an Event Source Mapping for the Lambda function by default, which means you can investigate and consume (or not) the message as preferred.
+
+You can temporarily set the replay SQS queue as an Event Source Mapping for the Lambda function, which means messages in the queue will be consumed by the Lambda and ingestion retried for transient failures. If the failure persists, the affected log entry will be moved to a DLQ after three retries.
+
+Every other error that occurs during the execution of the Lambda function is silently ignored, and reported to the APM server if instrumentation is enabled.
+
+### Execution timeout
+There is a grace period of 2 minutes before the timeout of the Lambda function where no more ingestion will occur. Instead, during this grace period the Lambda will collect and handle any unprocessed payloads in the batch of the input used as trigger.
 
-For errors at (2) the situation is different: we have now a state for N failed events out of total X events, if we fail the whole Lambda all the X events will be processed again. While the N failed ones could now succeed, the remaining X-N will now fail, since the datastreams are append-only and we would try to recreate already ingested documents (the ID of the document is deterministic).
+For CloudWatch Logs event input, S3 SQS Event Notifications input and direct SQS message payload input, the unprocessed batch will be sent to the SQS continuing queue.
 
-In case any of these errors will happen the Lambda won't return a failure. However, the payload of the event that failed to be ingested will be sent to a replay SQS queue.
-The replay SQS queue is set up by the Lambda deployment automatically.
-The replay SQS queue is not set as Event Source Mapping for the Lambda by default: the user can consume the message as preferred in order to investigate the failure.
-It is possible anyway to temporarily set the replay SQS queue as Event Source Mapping for the Lambda: in this case the messages in the queue will be consumed by the Lambda and tried to be ingested again if the failure was transient.
-In case the failure will persist the affected log entry will be moved to a DLQ after three retries.
+For Kinesis Data Stream input, the Lambda function will return the sequence numbers of the unprocessed batch in the `batchItemFailures` response, which allows the affected records to be included in later batches that will trigger the Lambda. It is therefore important to set the number of retry attempts high enough, and/or lower the batch size, to ensure the whole batch can be processed either during a single execution of the function or, with extra retry attempts, during multiple executions of the function.
```

Comment:
`Kinesis Data Stream input` is reworded as `Kinesis data stream` at https://github.com/elastic/elastic-serverless-forwarder/pull/131/files#diff-e9ecc3d1058b2f97b23c473ba67681d3e8e5c20546a6a7157b3bf9c1fc3c2e9cR25